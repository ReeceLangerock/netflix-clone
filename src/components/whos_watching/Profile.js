import PropTypes from 'prop-types';
import React from 'react'
import styled from 'styled-components'
import sg from './../../styleguide.js'

import bg from './../../assets/rick.jpg'
import meeseeks from './../../assets/meeseeks.jpg'
// import morty from './../../assets/morty.png'

const Avatar = styled.div`
height: 10vw;
width: 10vw;
max-height: 200px;
max-width: 200px;
min-height: 84px;
min-width: 84px;
border: .3em solid rgba(0,0,0,.4);
box-sizing: border-box;
position: relative;
text-decoration: none;
background-image: url(${props => (props.img ? require('./../../assets/' + props.img) : 'none')});
background-repeat: no-repeat;
background-size: contain;
background-color: ${props => (props.img ? '#333' : '')};
border: ${props => (props.img ? '.3em solid rgba(0,0,0,.4)' : 'none')};

display: flex;
justify-content: center;
align-items: center;
i{font-size: 6vw;
}

`

// ${props => 'url'+(require('./../../assets/' + props.userName.toLowerCase() + '.png'))};
const Container = styled.div`
margin: 0 2vw 0 0;
display: flex;
flex-direction: column;
align-content: center;
color: grey;
:last-child{
    margin: 0;
}

:hover {      
      ${Avatar}  {
        background-color: #e5e5e5;
        border-color: #e5e5e5;
        
        i {
          color: grey;
        }
        }
        
    }


`
const Name = styled.div`
line-height: 1.2em;
min-height: 2.4em;
display: block;
text-align: center;
font-size: 1.3vw;
margin: .8em 0;
text-overflow: ellipsis;
overflow: hidden;


`

export default class Profile extends React.Component {
  componentWillMount () {}
  render () {
    return (
      <Container>
        <Avatar img={this.props.img}>
          {!this.props.img && <i className={`fa fa-plus-circle`} aria-hidden='true' />}
        </Avatar>
        <Name>{this.props.userName}</Name>

      </Container>
    )
  }
}

Profile.propTypes = {
  userName: PropTypes.string.isRequired,
  img: PropTypes.string
};