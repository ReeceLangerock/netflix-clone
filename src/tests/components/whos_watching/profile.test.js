import React from 'react'
import ReactDOM from 'react-dom'
import Profile from './../../../components/whos_watching/Profile'

describe('Profile', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Profile userName='name' />, div)
  })

  it('should receive a name prop', () => {
    const wrapper = mount(<Profile userName='name' />)
    expect(wrapper.prop('userName')).toEqual('name')
    // expect(component.prop('img')).toEqual('img.png')
  })
  it('should render a <i> if no image prop', () => {
    const wrapper = mount(<Profile userName='name' />)
    expect(wrapper.find('div').find('i').length).toBe(1)
  })

  it('should not render a <i> image is included', () => {
    const wrapper = mount(<Profile userName='name' img='meeseeks.jpg' />)
    expect(wrapper.find('div').find('i').length).toBe(0)
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<Profile userName='name' />)
    expect(wrapper).toMatchSnapshot()
  })
})
