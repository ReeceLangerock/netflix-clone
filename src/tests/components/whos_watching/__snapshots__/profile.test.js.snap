// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    userName="name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div
          img={undefined}
        >
          <i
            aria-hidden="true"
            className="fa fa-plus-circle"
          />
        </styled.div>,
        <styled.div>
          name
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <i
            aria-hidden="true"
            className="fa fa-plus-circle"
          />,
          "img": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "className": "fa fa-plus-circle",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "name",
        },
        "ref": null,
        "rendered": "name",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            img={undefined}
          >
            <i
              aria-hidden="true"
              className="fa fa-plus-circle"
            />
          </styled.div>,
          <styled.div>
            name
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <i
              aria-hidden="true"
              className="fa fa-plus-circle"
            />,
            "img": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": "fa fa-plus-circle",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "name",
          },
          "ref": null,
          "rendered": "name",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
